src/java/org/apache/cassandra/io/util/RandomAccessReader.java

```diff
@@ -94,80 +93,96 @@ public ByteOrder order()
-        long position = getPosition();
-
-        FloatBuffer floatBuffer;
-        if (bh.offset() == 0 && position % Float.BYTES == 0 && bh.order() == order)
-        {
-            // this is a separate code path because buffer() and asFloatBuffer() both allocate
-            // new and relatively expensive xBuffer objects, so we want to avoid doing that
-            // twice, where possible. If the BufferHandler has a different underlying
-            // byte order, we duplicate first because there is not yet a way to configure
-            // the buffer handler to use the correct byte order.
-            floatBuffer = bh.floatBuffer();
-            floatBuffer.position(Ints.checkedCast(position / Float.BYTES));
-        }
-        else
+        var copied = 0;
+        while (copied < count)
         {
-            // bufferHolder offset is non-zero, and probably not aligned to Float.BYTES, so
-            // set the position before converting to FloatBuffer.
-            var bb = bh.buffer();
-            bb.order(order);
-            bb.position(Ints.checkedCast(position - bh.offset()));
-            floatBuffer = bb.asFloatBuffer();
-        }
+            var bh = bufferHolder;
+            long position = getPosition();
 
-        int bytesToRead = Float.BYTES * count;
-        if (count > floatBuffer.remaining())
-        {
-            // slow path -- desired slice is across region boundaries
-            var bb = getTemporaryBuffer(bytesToRead);
-            readFully(bb);
-            floatBuffer = bb.asFloatBuffer();
-        }
+            FloatBuffer floatBuffer;
+            if (bh.offset() == 0 && position % Float.BYTES == 0 && bh.order() == order)
+            {
+                // this is a separate code path because buffer() and asFloatBuffer() both allocate
+                // new and relatively expensive xBuffer objects, so we want to avoid doing that
+                // twice, where possible. If the BufferHandler has a different underlying
+                // byte order, we duplicate first because there is not yet a way to configure
+                // the buffer handler to use the correct byte order.
+                floatBuffer = bh.floatBuffer();
+                floatBuffer.position(Ints.checkedCast(position / Float.BYTES));
+            }
+            else
+            {
+                // bufferHolder offset is non-zero, and probably not aligned to Float.BYTES, so
+                // set the position before converting to FloatBuffer.
+                var bb = bh.buffer();
+                bb.order(order);
+                bb.position(Ints.checkedCast(position - bh.offset()));
+                floatBuffer = bb.asFloatBuffer();
+            }
 
-        floatBuffer.get(dest, offset, count);
-        seek(position + bytesToRead);
+            var remaining = floatBuffer.remaining();
+            if (remaining == 0)
+            {
+                // slow path -- desired slice is across region boundaries
```

Comment:
Nit: I'd amend that comment a bit. It's still a slower path, but "desired slice is across region boundaries" is, while not strictly incorrect for some definition of "slice", at least unnecessary imprecise now. And it's a good opportunity to give a few details. My suggestion would go something like (but feel free to amend/shorten):
```
// slow path -- the next float bytes are across a buffer boundary (we never start a loop iteration with the
// "current" buffer fully exhausted, so `remaining == 0` truly means "some bytes remains, but not
// enough for a float"), so we read that float individually (which will read it byte by byte, reBuffering as
// needed). After that we loop, which will switch back to the faster path for any remaining floats in the
// newly reloaded buffer.
```

Reply:
Great suggestion, you're right that the comment can be improved.